src/drivers/model/src/lib.rs

```diff
@@ -20,4 +19,16 @@ pub trait Driver {
     fn pwrite(&mut self, data: &[u8], pos: usize) -> Result<usize>;
     /// Shutdown the device.
     fn shutdown(&mut self);
+    /// Reads the exact number of bytes to fill in the `data`.
+    /// Returns ok if `data` is empty.
+    fn pread_exact(&self, data: &mut [u8], pos: usize) -> Result<()> {
+        let mut r = 0;
+        while r < data.len() {
```

Comment:
Is it possible to read more data than you asked for?

Reply:
No, unless the pread implementation is invalid.

Reply:
Sounds good. Just wanted you to think about it.